src/pipeline.py

```diff
@@ -45,6 +45,12 @@ def run_pipeline(user_config_path, dev_config_path):
     # validate the filenames in the config
     config = filename_validation(config)
 
+    # Check validate survey config
+    config = validate_survey_config(config)
+
+    # Check if PNP is set and config is valid
+    config = validate_pnp_config(config)
+
```

Comment:
to keep pipeline.py as simple as possible, could these two new functions be combined with filename_validation, do you think? You might like to give filename_validation a new name. 

By the way, the reason we can't do this validation when we actually set up the config is because we need to do it after the logger has been initialised.